packages/cli/src/lib/setup.ts

```diff
@@ -581,25 +523,31 @@ ${this.generateUserSetupFunctions()}`;
 # Task ID: ${this.taskId}
 # Task description is mounted at /task/description.json
 
-uid=$1
-gid=$2
+set -x
 
-echo "UID is $uid"
-echo "GID is $gid"
+if [[ -z "\${HOME}" ]]; then
+  export HOME=/home/agent
+fi
 
 # Some tools might be installed under /root/local/.bin conditionally
 # depending on the chosen agent and requirements, make this directory
 # available in the $PATH
 export PATH=/root/local/.bin:$PATH
 
+sudo mkdir -p $HOME
+sudo chown -R $(id -u):$(id -g) $HOME
```

Comment:
For example, can we replace these two lines with?

```suggestion
mkdir -p $HOME
```

Or do we need the second one for rootless?